### PR TITLE
Add a password strength indicator on relevant forms

### PR DESCRIPTION
**Pull request by AntoineAugusti**
Implementation of https://github.com/cachethq/Cachet/issues/940

Frontend inspired by https://github.com/kabum/strengthify and https://github.com/dropbox/zxcvbn is used for the password strength. Password strength is scored between 0 and 4 (as an integer), the UI will mark the password as "good" for a score of 3 or 4.

Pages done :
- First administrator setup
- Adding a new user to the team
- Editing a profile
- Listing users of the team

Screenshots:
- http://puu.sh/kvZpK/2db4da91b3.png
- http://puu.sh/kvZqi/9acf540161.png
- http://puu.sh/kvZsY/c4dad7707e.png
- http://puu.sh/kvZA6/b78136e73a.png

### CODE REVIEW

**Comment by TheoBearman** [comment]
:thumbsup:

**Comment by jbrooksuk** [comment]
This is awesome!

**Comment by billmn** [comment]
Nice!

**Comment by jbrooksuk** [comment]
Are you happy with this @AntoineAugusti? Looks fine to me!

**Comment by AntoineAugusti** [comment]
@jbrooksuk I think I am! :shipit:

**Comment by joecohens** [comment]
Can we include the password library via bower?

```
bower search strengthify
Search results:

    strengthify git://github.com/MorrisJobke/strengthify.git
```

**Comment by AntoineAugusti** [comment]
@joecohens The JS and the CSS have been tweaked because it does not suit the need. The JS assumes we are going to display a label + the coloured bar, and the CSS was a bit broken :)

File: resources/assets/sass/plugins/_password-strength.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+.strengthify-separator {
+    display: inline-block;
+    position: absolute;
+    background-color: #FFF;
+    width: 1px;
+    z-index: 10;
+}
+
+.password-bad {
+    background-color: #C33;
```
Comment (joecohens) [inline resources/assets/sass/plugins/_password-strength.scss]: @AntoineAugusti Can you use our palette colours? https://github.com/cachethq/Cachet/blob/master/resources/assets/sass/_palette.scss

Comment (joecohens) [inline resources/assets/sass/plugins/_password-strength.scss]: Also in the medium and good colours please :)

**Comment by joecohens** [comment]
@AntoineAugusti Oh ok so the bower thing never mind :)

**Comment by AntoineAugusti** [comment]
@joecohens Done and squashed ;)

**Comment by joecohens** [comment]
Awesome thanks :)

**Comment by jbrooksuk** [comment]
Sweet!

**Pull request merged**